File: CV/MP4/p4/code/evalBlobsDetection.py
import os
import numpy as np
import matplotlib.pyplot as plt
from utils import imread
from detectBlobs import detectBlobs
from drawBlobs import drawBlobs

# Evaluation code for blob detection
# Your goal is to implement scale space blob detection using LoG  
#
# This code is part of:
#
#   CMPSCI 670: Computer Vision
#   University of Massachusetts, Amherst
#   Instructor: Subhransu Maji

imageName = 'butterfly.jpg'
numBlobsToDraw = 500
imName = imageName.split('.')[0]

datadir = os.path.join('..', 'data', 'blobs')
im = imread(os.path.join(datadir, imageName))

blobs = detectBlobs(im)  # dummy placeholder

drawBlobs(im, blobs, numBlobsToDraw)

File: CV/MP4/p4/code/evalBlobsDetection.m
% Evaluation code for blob detection
% Your goal is to implement scale space blob detection using LoG 
%
% This code is part of:
%
%   CMPSCI 670: Computer Vision
%   University of Massachusetts, Amherst
%   Instructor: Subhransu Maji

imageName = 'butterfly.jpg';
numBlobsToDraw = 500;
imName = imageName(1:end-4);

dataDir = fullfile('..','data','blobs');
im = imread(fullfile(dataDir, imageName));


%% Implement the code to detect blobs here
blobs = detectBlobs(im); % dummy placeholder

%% Draw blobs on the image
drawBlobs(im, blobs, numBlobsToDraw);
title('Blob detection');
